src/dataset/custom_dataset.py

```diff
@@ -35,9 +35,10 @@ def __init__(self, args):
         self.total_episodes = 0
         self.seed_finder = SeedFinder()
         self.level = self.args["level"]
+        self.category = self._get_category()
         self.train_config, self.test_configs = self._get_configs()
         self.train_seeds = self._get_train_seeds()
-        self.category = self._get_category()
+        self.max_steps = self._get_level_max_steps()
```

Comment:
setting max steps on init as this is being used repeatedly throughout the code and needs to be accessible upon loading